sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -943,36 +935,51 @@ public Mono<VerifyResult> verifyData(SignatureAlgorithm algorithm, byte[] data,
 
     private Mono<Boolean> isValidKeyLocallyAvailable() {
         if (localOperationNotSupported) {
-            return Mono.just(false);
+            return Mono.defer(() -> Mono.just(false));
+        } else if (this.key == null && implClient.getKeyCollection() != null) {
+            // Get key from service and validate it. Then attempt to create a local cryptography client or default to
+            // service.
+            Mono<JsonWebKey> jsonWebKeyMono = CryptographyUtils.SECRETS_COLLECTION.equals(implClient.getKeyCollection())
```

Comment:
In regard to my previous comment, it seems this doesn't throw, right? Can you use this in your sync stack or something to avoid throwing? I don't know enough about Mono, but handling 403 or 404 explicitly instead of throwing seems best.

Reply:
Calling `implClient.getSecretKey()` or `implClient.getKey()` will ultimately throw an `HttpResponseException` or a child exception when the operation is not successful